open-metadata-resources/open-metadata-deployment/charts/odpi-egeria-lab/templates/egeria-core.yaml

```diff
@@ -57,7 +57,7 @@ spec:
           image: "{{ if (.Values.image.egeria.registry | default .Values.imageDefaults.registry) }}{{ .Values.image.egeria.registry | default .Values.imageDefaults.registry }}/{{ end }}\
                   {{ if (.Values.image.egeria.namespace | default .Values.imageDefaults.namespace) }}{{ .Values.image.egeria.namespace | default .Values.imageDefaults.namespace }}/{{ end }}\
                   {{ .Values.image.egeria.name }}\
-                  :{{ .Values.image.egeria.tag | default .Values.imageDefaults.tag }}"
+                  :{{ .Values.image.egeria.tag | default .Values.egeria.version }}"
```

Comment:
Is this the right way around?  I would expect to take `.Values.image.egeria.version` and only if that is empty to default to the default tag (which I would expect would always be `latest`)...

Even then, though, we need to be careful here keeping in mind the specific semantics of `latest` and the pull policy (ie. this makes it further critical that the default pull policy remains `Always` so that non-`latest` images will be forcibly replaced by k8s and it won't leave stale / old `x.x-SNAPSHOT` images around)...

Reply:
On the version choice:

It was my intent to do it this way around. 

My reasoning - it is generally sensible only to use the image version that is the same as that of the chart. We ship that as 1.5-SNAPSHOT. Since we're moving to a more release-specific model I do not believe 'latest' makes so much sense. This is why the expression no longer uses the imageDefaults for the tag.  However I still wanted to allow the specific image version to be overriden on demand which can optionally be set.

Reply:
I did anticipate the second part -- the vdc chart does now use 'Always' as the default setting for exactly that reason.

I did consider whether to add an expression to use 'Always' for snapshots and 'ifNotPresent' for release, but was keen to make significant steps forward in the general aspects around release specific images, whilst maintaining compatability with our existing charts - even though vdc for example, is somewhat superceeded by events and will be reworked around the 1.8 time so wanted to move forward. We can refine the setting of that value through an additional issue/PR if you feel it would be helpful?

The whole intent of this change is to make -SNAPSHOT something only us devs on the team use, whilst the 'masses' use specific released versions. 

As a mitigation we could change the flag in values as the point we cut a release ? That vaguely crossed my mind.

Reply:
Fine for now, just need to keep in mind for down the road...